jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/MaximalCliqueEnumerationAlgorithm.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (C) Copyright 2017-2017, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.util.Iterator;
+import java.util.Set;
+
+/**
+ * A maximal clique enumeration algorithm.
+ * 
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ * 
+ * @author Dimitrios Michail
+ */
+public interface MaximalCliqueEnumerationAlgorithm<V, E>
+    extends Iterable<Set<V>>
+{
+
+    /**
+     * Returns an iterator over all maximal cliques.
+     *
+     * @return an iterator over all maximal cliques
+     */
+    @Override
+    Iterator<Set<V>> iterator();
+
+    /**
+     * Returns an iterator over all maximum cliques.
+     * 
+     * <p>
+     * This procedure just filters the maximal cliques by clique size.
+     * 
+     * @return an iterator over all the maximum cliques
+     */
+    Iterator<Set<V>> maximumIterator();
```

Comment:
Removed it from the interface, as it was indeed confusing. I left it in the implementation with improved documentation.